LIS3DH.py

```diff
@@ -137,7 +143,7 @@ def getAxis(self, axis):
       if range==self.RANGE_2G:    divisor = 16380
       elif range==self.RANGE_4G:  divisor = 8190
       elif range==self.RANGE_8G:  divisor = 4096
-      elif range==self.RANGE_16G: divisor = 2048
+      elif range==self.RANGE_16G: divisor = 1365.33
```

Comment:
Hi Matt

Table 3 in the data sheet has the sensitivity of 12 mg/digit for the 16G range compared to 1, 2 & 4 respectively for 2G, 4G, 8G
2048 would be the correct divisor if this was 8 mg/digit following the sequence but that’s not correct.
1365.33 is calculated as 4096 / 3

This checks out with measurements.  If you test out each range separately with the sensor static on the desk i get readings of circa 1g for the 2G, 4G & 8G readings and 0.66g for the 16G with the original value.
Altering this value as above makes the 16G range also read 1g

Sorry about the tabs!.  I’ll fix that.

All the best.

Mal



On 19 Mar 2016, at 22:17, Matt Dyson <notifications@github.com> wrote:

> In LIS3DH.py:
> 
> > @@ -137,7 +143,7 @@ def getAxis(self, axis):
> >        if range==self.RANGE_2G:    divisor = 16380
> >        elif range==self.RANGE_4G:  divisor = 8190
> >        elif range==self.RANGE_8G:  divisor = 4096
> > -      elif range==self.RANGE_16G: divisor = 2048
> > +      elif range==self.RANGE_16G: divisor = 1365.33
> Where did you get this number from? I was going off this which seems to indicate that 2048 is the correct divisor for 16G?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>